### ISSUE TITLE
`import-self` false positive when importing non-existing module from current package

### ISSUE THREAD

**Issue by vlad0337187** [PyCQA/pylint#3748]
### Steps to reproduce
1. create some module
2. try to import non-existing module from current package in `__init__.py`

### Current behavior
- raises `W0406: Module import itself (import-self)`

### Expected behavior
- should raise something like `X0000: Module import not found (import-missing)`

### pylint --version output
```
pylint 2.5.3
astroid 2.4.2
Python 3.8.3 (default, May 29 2020, 00:00:00) 
[GCC 10.1.1 20200507 (Red Hat 10.1.1-1)]
```

### Reproduce case
- download this .zip: [pylint_false_positive_import_self.zip](https://github.com/PyCQA/pylint/files/4977028/pylint_false_positive_import_self.zip)
- unpack it
- make unpacked directory - current (cd command)
- run `pipenv install`  (to install pylint, create venv)
- run `pipenv run pylint ./`

**Comment by dubiousjim** [comment]
Related issues: #3665, #5151, #6535, #7289
